### PR TITLE
saltpack: factor out "branding"

### PR DESCRIPTION
**Pull request by maxtaco**
- the word "KEYBASE" doesn't appear in code in the saltpack/ library
- it's supplied as a parameter to encryption, and a return value from decryption
  (and similarly for signing/verifying)
- should make it easy to adopt for other uses
- add more tests for bad frames

### CODE REVIEW

**Comment by maxtaco** [comment]
@patrickxb  / @oconnor663; it might be good if you both took a look...

**Comment by patrickxb** [comment]
So do we really want to error with checkSaltpackBrand?  

If someone writes some app that uses saltpack (with their own brand [or no brand]) and has support for keybase keys and they send an encrypted file to a keybase user, seems like using `keybase decrypt` should still work even if the brand is different.

**Comment by maxtaco** [comment]
the general tack has been to be most restrictive and loosen up in the future, since we can't really go the other way.

**Comment by patrickxb** [comment]
Ok.  LGTM.

File: go/libkb/saltpack.go [hunk 1]
```diff
@@ -137,3 +137,10 @@ func BoxPublicKeyToKeybaseKID(k saltpack.BoxPublicKey) (ret keybase1.KID) {
 	p := k.ToKID()
 	return keybase1.KIDFromRawKey(p, KIDNaclDH)
 }
+
+func checkSaltpackBrand(b string) error {
+	if b != KeybaseSaltpackBrand {
+		return KeybaseSaltpackError{}
+	}
+	return nil
+}
```
Comment (oconnor663) [inline go/libkb/saltpack.go]: Does it always make sense to error out on brands other than Keybase? Is it possible that someone might want to change the brand name but remain compatible with Keybase? (Or is that something we would want to discourage?)

Comment (maxtaco) [inline go/libkb/saltpack.go]: @patrickxb asked more or less the same question. I figured it's better to make it more restrictive at first and then loosen

Comment (oconnor663) [inline go/libkb/saltpack.go]: Ah, just saw Patrick's comment to the same effect.

**Comment by oconnor663** [comment]
The code looks good. Do we have a plan for revisiting this to decide whether to open it up? I worry that there's no natural decision point, other than now.

**Pull request merged**